genesis/utils/mjcf.py

```diff
@@ -1,15 +1,25 @@
 import os
 import xml.etree.ElementTree as ET
-from contextlib import redirect_stderr
+import contextlib
 from pathlib import Path
 from itertools import chain
 from bisect import bisect_right
+import io
 
 import numpy as np
 import trimesh
 from trimesh.visual.texture import TextureVisuals
 from PIL import Image
 
+if os.name != "nt":  # POSIX → keep current behaviour
```

Comment:
You should have a cleaner uniform way of redirecting libc stdout / stderr. I suggest to move all the logic here along with what I already implemented for macOS at least, and use the same cross platform decorator everywhere

Reply:
https://github.com/Genesis-Embodied-AI/Genesis/blob/6ec00a7410b46d46b5ab2b294ac5f8af6396b6bd/genesis/utils/misc.py#L47